### PR TITLE
[multi-asic][vs]: Update topology script to retrieve hwsku from minigraph

### PR DESCRIPTION
**Pull request by SuvarnaMeenakshi**
Update topology script to retrieve hwsku from minigraph
if hwsku information is not available in config_db.
Fix clean up of interfaces in msft_multi_asic_vs hwsku
topology script.

Signed-off-by: Suvarna Meenakshi <sumeenak@microsoft.com>

**- Why I did it**
When bringing up multi-asic VS switch, topology service is started during boot up.
Topology service starts a shell script which runs the topology script present in /usr/share/sonic/device/<platfrom>/<hwsku> directory. To invoke hwsku specific script, the topology script tries to retrieve hwsku information from config_db.
During initial boot up config_db might not be populated. In order to start topology service before config_db is updated,
update topology script to get hwsku information from minigraph.xml if it is available.
This will be helpful to bring up multi-asic VS testbed by loading minigraph and  starting topology service.

**- How I did it**
- Update topology.sh script to retrieve hwsku information from minigraph.xml.
- Fix clean up function on msft_multi_asic_vs toplogy script.

**- How to verify it**
- single-asic VS - no change; topology service is only enabled for multi-asic VS.
- multi-asic VS - Bring up multi-asic VS image, copy minigraph to vs image, start topology service. Topology service should be successful.
-  to test clean up function fix, start topology service - make sure interfaces are created and moved to the right namespaces.
stop topology service - make sure namespace do not have any interface and all front end interfaces are present in default namespace.

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [x] 201911
- [ ] 202006

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: device/virtual/x86_64-kvm_x86_64-r0/msft_multi_asic_vs/topology.sh [hunk 1]
```diff
@@ -8,10 +8,10 @@ NUM_INTERFACES_PER_ASIC=32
...
-    # eth16 - eth31: asic3 
-    # eth32 - eth47: asic4
-    # eth48 - eth63: asic5
+    # eth1  - eth16: asic0
+    # eth17 - eth32: asic1 
+    # eth33 - eth48: asic2
+    # eth49 - eth64: asic3
     for ASIC in `seq $FIRST_FRONTEND_ASIC $LAST_FRONTEND_ASIC`; do
         for NUM in `seq 1 16`; do
             ORIG="eth$((16 * $ASIC + $NUM))"
```
Comment (judyjoseph) [inline device/virtual/x86_64-kvm_x86_64-r0/msft_multi_asic_vs/topology.sh:17]: Looks good, had a question of whether we need to really rename the interface to eth1-eth16 in all the namespaces ? We could have moved the interface as is ? Was there some issue with libvssai if we did that ?

Comment (SuvarnaMeenakshi) [reply device/virtual/x86_64-kvm_x86_64-r0/msft_multi_asic_vs/topology.sh:17]: yes, vs has a lanemap.ini file which uses the interface names eth1 , eth2.. and maps each interface to lanes as defined in port_config.ini.

**Comment by SuvarnaMeenakshi** [comment]
retest this please

**Comment by SuvarnaMeenakshi** [comment]
retest this please

**Comment by SuvarnaMeenakshi** [comment]
retest vsimage please

**Comment by SuvarnaMeenakshi** [comment]
retest vsimage please

**Comment by SuvarnaMeenakshi** [comment]
retest this please

File: files/image_config/topology/topology.sh [hunk 2]
```diff
@@ -7,15 +7,33 @@
...
     /usr/share/sonic/device/$PLATFORM/$HWSKU/$TOPOLOGY_SCRIPT start
 }
 stop() {
     TOPOLOGY_SCRIPT="topology.sh"
-    PLATFORM=${PLATFORM:-`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`}
-    HWSKU=${HWSKU:-`sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]["hwsku"]'`}
-    usr/share/sonic/device/$PLATFORM/$HWSKU/$TOPOLOGY_SCRIPT stop 
+    PLATFORM=`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`
+    HWSKU=${HWSKU:-`sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]["hwsku"]' 2>&1`}
+    if [[ $? -ne 0 ]]; then
```
Comment (judyjoseph) [inline files/image_config/topology/topology.sh:27]: Confirm if we need to check for HWSKU == "" as the return value $? would be zero ?

Comment (SuvarnaMeenakshi) [reply files/image_config/topology/topology.sh]: Thank you, I have added that check here to ensure that we record all failure conditions.
Also added another check to exit, if HWSKU is not found in config_db and minigraph.xml file is also not available.

**Review by judyjoseph — APPROVED** [review]

File: files/image_config/topology/topology.sh [hunk 3]
```diff
@@ -7,15 +7,39 @@
...
 stop() {
     TOPOLOGY_SCRIPT="topology.sh"
-    PLATFORM=${PLATFORM:-`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`}
-    HWSKU=${HWSKU:-`sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]["hwsku"]'`}
-    usr/share/sonic/device/$PLATFORM/$HWSKU/$TOPOLOGY_SCRIPT stop 
+    PLATFORM=`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`
+    HWSKU=`sonic-cfggen -d -v 'DEVICE_METADATA["localhost"]["hwsku"]' 2>&1`
+    if [[ $? -ne 0 || HWSKU == "" ]]; then
+            if [[ -f "/etc/sonic/minigraph.xml" ]]; then
+                HWSKU=`sonic-cfggen -m /etc/sonic/minigraph.xml -v "DEVICE_METADATA['localhost']['hwsku']" 2>&1`
```
Comment (abdosi) [inline files/image_config/topology/topology.sh:32]: we can add this logic in function instead of duplicating it for start/stop

Comment (SuvarnaMeenakshi) [reply files/image_config/topology/topology.sh]: Updated as suggested.

**Review by abdosi — APPROVED** [review]

**Review by judyjoseph — APPROVED** [review]

**Pull request merged**